### ISSUE TITLE
Skript wont install

### ISSUE THREAD

**Issue by renegade2k82** [RetroFlag/retroflag-picase#126]
For the life of me i cannot get the skript to run on my ubuntu and this really sucks because i like the case alot.

**Comment by Michael-Zinn** [comment]
@renegade2k82 You should not run the script on your Ubuntu. You should run it on RetroPie or Recalbox on the pi (zero).
If you are running Ubuntu on a pi (zero) in a retroflag case, you will have to rewrite the script for Ubuntu yourself.

**Comment by renegade2k82** [comment]
I have a pi 4b 8gb.so are you saying i cant use the safe shutdown feature on ubuntu?Do i have to have that script to use the buttons on the case?

**Comment by zero4281** [comment]
> I have a pi 4b 8gb.so are you saying i cant use the safe shutdown feature on ubuntu?Do i have to have that script to use the buttons on the case?

The script has to be run on the RaspberryPi with RetroPie installed on the SD Card.  I don't think that Ubuntu is supported.

**Comment by renegade2k82** [comment]
Oh ok so if i run the script with retropie then switch back to ubuntu will the skirpt still take effect or do i have to have retropie installed forever for the skirpt to work?

**Comment by zero4281** [comment]
If you replace the operating system then the script won't be installed anymore.  You have to use RetroPie or customize the script to run with Ubuntu on your own.

**Comment by renegade2k82** [comment]
ok well can  i just set the safe shutdown switch to the off position and still use the case buttons to turn it on and off and to restart?

**Comment by zero4281** [comment]
> ok well can i just set the safe shutdown switch to the off position and still use the case buttons to turn it on and off and to restart?

That will do an unsafe shutdown that could corrupt the SD card.

**Comment by renegade2k82** [comment]
Well i guess i will be returning these retroflag cases then its a huge let down that you have to use the safe shutdown just to not corrupt anything and also im disappointed thats its support is very slim.not everyone uses retropie and the other ones.

**Comment by zero4281** [comment]
P.S. The issue with corrupting the SD card isn't because of the case.  The Raspberry Pi is always subject to that issue if you pull the plug on it.

**Comment by renegade2k82** [comment]
Yeah i have used retropie before i will say it is very nice and runs well but i only want to use my pi to host minecraft servers so thats why i hoped there was some way for the safe shutdown to work on ubuntu.would this work https://github.com/crcerror/retroflag-picase/pull/7/commits/128b311a1dbd9cb27bedf2abbf545d3a6e9042d2

**Comment by zero4281** [comment]
> Yeah i have used retropie before i will say it is very nice and runs well but i only want to use my pi to host minecraft servers so thats why i hoped there was some way for the safe shutdown to work on ubuntu.would this work [crcerror@128b311](https://github.com/crcerror/retroflag-picase/commit/128b311a1dbd9cb27bedf2abbf545d3a6e9042d2)

It's worth a shot dude.  It's a nice case, but it is a little bit of a DIY project.

**Comment by alanbchristie** [comment]
If you want to run on Ubuntu you can. I'm using the 64-bit 21.04 on an 8Gi RPi 4 but had a number of problems. The shutdown script is broken anyway (#125) but that's an easy fix. In the end I created some notes for my installation (just in case I needed to do it again). Don't know if they'd be of any use...

https://gist.github.com/alanbchristie/a421cb9568ac9611436e16dee75e589b
